LeoBotUrdf

LyubomyrD opened the pull request:
**Commit Summary:**

Added two standard wheels
Added one caster wheel

Closing #29

LyubomyrD committed:
LeoBotUrdf

AndriyPt commented on leobot_description/robot/robot.urdf.xacro:
```diff
@@ -14,6 +14,60 @@
     <xacro:cylinder_collision radius="0.25" height="1.0"/>
     <xacro:cylinder_inertia mass="10" radius="0.25" height="1.0"/>
   </link>
+       <!-- Leo_left_wheel -->
```
I guess comment left wheel for link with name left_wheel_link is redundant :)

AndriyPt commented on leobot_description/robot/robot.urdf.xacro:
```diff
@@ -14,6 +14,60 @@
     <xacro:cylinder_inertia mass="10" radius="0.25" height="1.0"/>
   </link>
+       <!-- Leo_left_wheel -->
+    <link name="left_wheel_link">
```
Please fix XML formatting

AndriyPt commented on leobot_description/robot/robot.urdf.xacro:
```diff
@@ -14,6 +14,60 @@
+    <link name="right_wheel_link">
+    <visual>
+      <xacro:cylinder_geometry radius="0.1" height="0.05"/>
+      <xacro:red_material/>
```
In Gazebo you are using White colour in URDF Red. It might mislead people. Please use one colour for links in Gazebo and URDF :)

AndriyPt commented on leobot_description/robot/robot.urdf.xacro:
```diff
@@ -30,6 +84,9 @@
     <child link="camera_left_link"/>
   </joint>
 
+
```
Please remove extra empty lines

AndriyPt commented on leobot_description/robot/robot.urdf.xacro:
```diff
@@ -14,6 +14,60 @@
+
+    <!-- left_wheel-body -->
+  <joint name="left_wheel_body_joint" type="continuous">
+    <origin rpy="1.570795 0 0" xyz="0.0 0.275 -0.55"/>
```
Pi is predefined constant. Please use XACRO expressions to calculate 90 degrees as here http://wiki.ros.org/xacro#Math_expressions

AndriyPt reviewed (changes requested):
Few remarks

LyubomyrD committed:
LeoBotUrdf

LyubomyrD commented:
@AndriyPt Thank You for comments. It seems that I corrected everything. Have another look, please.

AndriyPt commented:
Robot is not stable in Gazebo simulation
![image](https://user-images.githubusercontent.com/10820297/35874181-88f6a92c-0b74-11e8-8a7e-73758036dd07.png)

AndriyPt commented on leobot_description/robot/leobot.gazebo.xacro:
```diff
@@ -18,6 +18,33 @@
+    <kd value="1.0"/>
+    <material>Gazebo/Red</material>
+  </gazebo>
+
```
Please remove not needed empty lines

AndriyPt commented on leobot_description/robot/robot.urdf.xacro:
```diff
@@ -15,6 +15,54 @@
+    <parent link="base_link"/>
+    <child link="caster_wheel_link"/>
+  </joint>
+
```
Please remove not needed empty lines

AndriyPt commented:
@LyubomyrD description should be "Closing #<issue number>"

AndriyPt commented:
@LyubomyrD please also add transmission as it is done here https://github.com/AndriyPt/ros-navigation-examples/blob/kinetic-devel/moon_description/urdf/wheels.urdf.xacro#L57-L66

LyubomyrD commented:
@AndriyPt Thank You for You comments. 

I want to discuss the falling robot issue. 

I guess, that the easiest way is to increase the mass of the caster wheel. But I suggest doing this after @SystemDiagnosticss will finish the head of the robot. The robot is rather tall and all additional mass may cause the instability. 

Another option is that @SystemDiagnosticss set the head weight as almost zero, hence it will not have much influence on the stability of our "pendulum".

What do You think?

AndriyPt commented:
@LyubomyrD you could add wider platform on the bottom to hold the wheels and add extra stability. It might be cylinder with bigger radius.

AndriyPt commented:
@LyubomyrD I have merged @SystemDiagnosticss `head_link` to trunk. Please resolve URDF conflicts. Please let me know if you need assistance with that :)

LyubomyrD committed:
Resolved Conflicts

Closing #29

LyubomyrD committed:
Resolved Conflicts
Added transmission
Added wheel xacro macros
Added wheel-base bearing

Closing #29

LyubomyrD committed:
Merge branch 'kinetic-devel' into F#29_URDF_for_base_with_wheels

LyubomyrD committed:
Resolved Conflicts
Added transmission
Added wheel xacro macros
Added wheel-base bearing

Closing #29

AndriyPt commented:
PR looks really good.
The only remark is please return TF to RVIz configuration and happy to merge :)
![image](https://user-images.githubusercontent.com/10820297/35994513-51692498-0d19-11e8-8845-ac85b7d18663.png)

LyubomyrD committed:
Resolved Conflicts
Added transmission
Added wheel xacro macros
Added wheel-base bearing

Closing #29
